toolchain/lexer/tokenized_buffer_benchmark.cpp

```diff
@@ -89,41 +76,214 @@ auto GenerateRandomIdentifier(absl::BitGen& gen, int min_length = 1,
+      "than 64 as those are exceedingly rare in our observed data sets.");
+
+  static const std::array<std::string, NumTokens> id_storage = [] {
+    std::array<int, 64> id_length_counts;
+    // For non-uniform distribution, we simulate a distribution roughly based on
+    // the observed histogram of identifier lengths in LLVM, but smoothed a bit
+    // and reduced to small counts so that we cycle through all the lengths
+    // reasonably quickly. We want the sampling even 10% of NumTokens from this
+    // in a round-robin form to not be skewed overly much. This still inherently
+    // compresses the long tail as we'd rather have coverage even though it
+    // distorts the distribution a bit.
+    //
+    // 1 characters   [3976]  ███████████████████████████████▊
+    id_length_counts[0] = 40;
+    // 2 characters   [3724]  █████████████████████████████▊
+    id_length_counts[1] = 40;
+    // 3 characters   [4173]  █████████████████████████████████▍
+    id_length_counts[2] = 40;
+    // 4 characters   [5000]  ████████████████████████████████████████
+    id_length_counts[3] = 50;
+    // 5 characters   [1568]  ████████████▌
+    id_length_counts[4] = 20;
+    // 6 characters   [2226]  █████████████████▊
+    id_length_counts[5] = 20;
+    // 7 characters   [2380]  ███████████████████
+    id_length_counts[6] = 20;
+    // 8 characters   [1786]  ██████████████▎
+    id_length_counts[7] = 18;
+    // 9 characters   [1397]  ███████████▏
+    id_length_counts[8] = 12;
+    // 10 characters  [ 739]  █████▉
+    id_length_counts[9] = 12;
+    // 11 characters  [ 779]  ██████▎
+    id_length_counts[10] = 12;
+    // 12 characters  [1344]  ██████████▊
+    id_length_counts[11] = 12;
+    // 13 characters  [ 498]  ████
+    id_length_counts[12] = 5;
+    // 14 characters  [ 284]  ██▎
+    id_length_counts[13] = 3;
+    // 15 characters  [ 172]  █▍
+    // 16 characters  [ 278]  ██▎
+    // 17 characters  [ 191]  █▌
+    // 18 characters  [ 207]  █▋
+    for (int i : llvm::seq(14, 18)) {
+      id_length_counts[i] = 2;
+    }
+    // 19 - 63 characters are all <100 in LLVM but non-zero, and we map them to
+    // 1 for coverage despite slightly over weighting the tail.
+    for (int i : llvm::seq(18, 64)) {
+      id_length_counts[i] = 1;
+    }
+
+    // Used to track the different count buckets when in a non-uniform
+    // distribution.
+    int length_bucket_index = 0;
+    int length_count = 0;
+
+    std::array<std::string, NumTokens> ids;
+    absl::BitGen gen;
+    for (auto [i, id] : llvm::enumerate(ids)) {
+      if (Uniform) {
+        // Rather than using randomness, for a uniform distribution rotate
+        // lengths in round-robin to get a deterministic and exact size on every
+        // run. We will then shuffle them at the end to produce a random
+        // ordering.
+        int length = MinLength +
+                     (MaxLength > MinLength ? i % (MaxLength - MinLength) : 0);
```

Comment:
If MinLength == 3 and MaxLength == 5, you get `MinLength + i % (MaxLength - MinLength)` or `3 + i % 2`. This will result in values of 3 or 4. If MaxLength is intended to be non-inclusive, maybe you should just require MaxLength > MinLength and the 1-char case should be `1, 2` instead of `1, 1`? Otherwise, I think this could be just `MinLength + (i % (MaxLength - MinLength + 1))`, no ternary.